src/authentication/authentication.service.ts

```diff
@@ -16,11 +21,43 @@ export class AuthenticationService {
       const correctPassword = await user.checkPassword(password)
       return correctPassword ? user : undefined
     }
-    return undefined
   }
 
-  async login(user: User): Promise<string> {
+  async generateRefreshToken(user: User): Promise<string | undefined> {
+    try {
+      const session = new AuthSession()
+      session.refreshToken = this.generateNonce()
+      session.userId = user.id
+      await this.sessionRepo.create(session)
+      await this.sessionRepo.save(session)
+      return session.refreshToken
+    } catch {}
+  }
+
+  private generateNonce() {
```

Comment:
nit: this is also defined in src/oauth/oauth.service.ts - something like generating a secure nonce should only be defined in one place (don't want any typos or drift).  could put it in a utils/crypto.ts file or something.